mbed-trace/mbed_trace.h

```diff
@@ -83,46 +83,60 @@ extern "C" {
 /** Config mask */
 #define TRACE_MASK_CONFIG         0xE0
 /** Trace level mask */
-#define TRACE_MASK_LEVEL          0x1F
+#define TRACE_MASK_LEVEL          0x7F
 
 /** plain trace data instead of "headers" */
-#define TRACE_MODE_PLAIN          0x80
+#define TRACE_MODE_PLAIN          0x400
 /** color mode */
-#define TRACE_MODE_COLOR          0x40
+#define TRACE_MODE_COLOR          0x200
 /** Use print CR before trace line */
-#define TRACE_CARRIAGE_RETURN     0x20
+#define TRACE_CARRIAGE_RETURN     0x100
 
 /** used to activate all trace levels */
-#define TRACE_ACTIVE_LEVEL_ALL    0x1F
+#define TRACE_ACTIVE_LEVEL_ALL    0x7F
 /** print all traces same as above */
-#define TRACE_ACTIVE_LEVEL_DEBUG  0x1f
-/** print info,warn and error traces */
-#define TRACE_ACTIVE_LEVEL_INFO   0x0f
-/** print warn and error traces */
-#define TRACE_ACTIVE_LEVEL_WARN   0x07
-/** print only error trace */
-#define TRACE_ACTIVE_LEVEL_ERROR  0x03
+#define TRACE_ACTIVE_LEVEL_SILLY  0x7f
+/** print debug,warn, error and critical traces */
+#define TRACE_ACTIVE_LEVEL_DEBUG  0x3f
+/** print info, warn, error and critical traces */
+#define TRACE_ACTIVE_LEVEL_INFO   0x1f
+/** print warn, error and critical traces */
+#define TRACE_ACTIVE_LEVEL_WARN   0x0f
+/** print error and critical trace */
+#define TRACE_ACTIVE_LEVEL_ERROR  0x07
+/** print only critical trace */
+#define TRACE_ACTIVE_LEVEL_CRITICAL  0x03
 /** print only cmd line data */
 #define TRACE_ACTIVE_LEVEL_CMD    0x01
 /** trace nothing  */
 #define TRACE_ACTIVE_LEVEL_NONE   0x00
```

Comment:
Why do the trace levels need to be defined as bitmaps? Even with just the original 8 bits for the configuration, 5 bits for the trace level would suffice for 31 separate levels.

Reply:
heh, good question :). I can’t remember any good reason. Maybe we should change it now..

Reply:
Would be a smaller API change if it's only the addition of the tr_silly trace level and macros. Not sure what the most significant cons would be though.

Reply:
replace bitmask with number which coming from lower bits..